internal/cloud/awscloud/secure-instance.go

```diff
@@ -194,6 +199,42 @@ func (a *AWS) TerminateSecureInstance(si *SecureInstance) error {
 	return nil
 }
 
+func (a *AWS) terminatePreviousSI(hostInstanceID string) (string, error) {
+	descrInstancesOutput, err := a.ec2.DescribeInstances(&ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name:   aws.String("tag:parent"),
+				Values: []*string{aws.String(hostInstanceID)},
+			},
+		},
+	})
+	if err != nil {
+		return "", err
```

Comment:
… maybe change this for better error messages
```suggestion
		return "no instance found", err
```

Reply:
Well in this case the describe instances call failed, so it's not that there were no instances found.